### PR TITLE
drivers: adc: esp32: dma enable

### PR DESCRIPTION
**Pull request by wmrsouza**
adc dma operation mode on ESP32S3 and ESP32C3

### CODE REVIEW

File: drivers/adc/adc_esp32.c [hunk 1]
```diff
@@ -284,16 +603,76 @@ static int adc_esp32_channel_setup(const struct device *dev, const struct adc_ch
...
+	if (k_sem_init(&data->dma_conv_wait_lock, 0, 1)) {
+		LOG_ERR("dma_conv_wait_lock initialization failed!");
+		return -EINVAL;
+	}
+
+	data->adc_hal_dma_ctx.rx_desc = k_aligned_alloc(sizeof(uint32_t),
+							sizeof(dma_descriptor_t));
+	if (!data->adc_hal_dma_ctx.rx_desc) {
+		LOG_ERR("rx_desc allocation failed!");
+		return -EINVAL;
```
Comment (uLipe) [inline drivers/adc/adc_esp32.c:658]: nitpick I think -ENOMEM reflect better the error here.

File: drivers/adc/adc_esp32.c [hunk 2]
```diff
@@ -284,16 +603,76 @@ static int adc_esp32_channel_setup(const struct device *dev, const struct adc_ch
...
+	}
+	LOG_DBG("rx_desc = 0x%08X", (unsigned int)data->adc_hal_dma_ctx.rx_desc);
+
+	data->dma_buffer = k_aligned_alloc(sizeof(uint32_t), ADC_DMA_BUFFER_SIZE);
+	if (!data->dma_buffer) {
+		LOG_ERR("dma buffer allocation failed!");
+		k_free(data->adc_hal_dma_ctx.rx_desc);
+		return -EINVAL;
```
Comment (uLipe) [inline drivers/adc/adc_esp32.c:666]: Same for this one.

**Review by uLipe — DISMISSED** [review]
Just left some nitpicks, otherwise, looks great!

**Review by LucasTambor — APPROVED** [review]

**Review by uLipe — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Comment by wmrsouza** [comment]
@anangl could you please take a look?

File: drivers/adc/Kconfig.esp32 [hunk 3]
```diff
@@ -8,3 +8,15 @@ config ADC_ESP32
 	depends on DT_HAS_ESPRESSIF_ESP32_ADC_ENABLED
 	help
 	  Enable the driver implementation for the ESP32 ADC
+
+if ADC_ESP32
+
+config ADC_ESP32_DMA
+	bool "ESP32 ADC DMA Support"
+	default n
```
Comment (anangl) [inline drivers/adc/Kconfig.esp32:16]: `default n` is redundant.

**Review by anangl — APPROVED** [review]
"tests: drivers: adc" is not a very informative commit title

**Pull request merged**